iib/web/models.py

```diff
@@ -190,6 +209,104 @@ def add_state(self, state, state_reason):
         db.session.flush()
         self.request_state_id = request_state.id
 
+    def to_json(self, verbose=True):
+        def _state_to_json(state):
+            return {
+                'state': RequestStateMapping(state.state).name,
+                'state_reason': state.state_reason,
+                'updated': state.updated.isoformat() + 'Z',
+            }
+
+        rv = {
+            'id': self.id,
+            'arches': [arch.name for arch in getattr(self.index_image, 'architectures', [])],
+            'binary_image': self.binary_image.pull_specification,
+            'binary_image_resolved': getattr(
+                self.binary_image_resolved, 'pull_specification', None),
+            'bundles': [bundle.pull_specification for bundle in self.bundles],
+            'from_index': getattr(self.from_index, 'pull_specification', None),
+            'from_index_resolved': getattr(self.from_index_resolved, 'pull_specification', None),
+            'index_image': getattr(self.index_image, 'pull_specification', None),
+            'user': getattr(self.user, 'username', None),
+        }
+
+        latest_state = None
+        if verbose:
+            states = [_state_to_json(state) for state in self.states]
+            # Reverse the list since the latest states should be first
+            states = list(reversed(states))
+            rv['state_history'] = states
+            latest_state = states[0]
+        rv['state'] = latest_state or _state_to_json(self.state)
```

Comment:
We want "state", "state_reason" and "updated" to be on the top-level of the JSON object.

```suggestion
        rv.update(latest_state or _state_to_json(self.state))
```